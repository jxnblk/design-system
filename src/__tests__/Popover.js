import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { Popover } from '..'

const noop = () => {}

describe('Popover', () => {
  test('renders', () => {
    const json = renderer
      .create(<Popover open={false} onDismiss={noop} />)
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders open', () => {
    const json = renderer
      .create(<Popover open={true} onDismiss={noop} />)
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('Popover.Body renders', () => {
    const json = renderer.create(<Popover.Body />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('calls onDismiss callback on body click', () => {
    const onDismiss = jest.fn()
    const map = {}
    document.addEventListener = jest.fn((event, cb) => {
      map[event] = cb
    })
    const wrapper = mount(
      <div id="root">
        <Popover open={true} onDismiss={onDismiss} />
      </div>
    )
    map.click({})
    expect(onDismiss.mock.calls.length).toBe(1)
  })

  test('removes event listener on unmount', () => {
    const onDismiss = jest.fn()
    const map = {}
    document.addEventListener = jest.fn((event, cb) => {
      map[event] = cb
    })
    const wrapper = mount(
      <div id="root">
        <Popover open={true} onDismiss={onDismiss} />
      </div>
    )
    wrapper.unmount()
    map.click({})
    expect(onDismiss.mock.calls.length).toBe(0)
  })
})
