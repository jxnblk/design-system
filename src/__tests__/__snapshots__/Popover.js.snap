// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover Popover.Body renders 1`] = `
.c1 {
  margin-top: 0px;
  padding: 16px;
  background-color: #fff;
}

.c0 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow: auto;
  -webkit-overflow-scrolling: touch;
}

.c0 {
  position: absolute;
  left: 0;
}

@media screen and (min-width:40em) {
  .c1 {
    margin-top: 8px;
  }
}

<div
  className="c0 c1"
/>
`;

exports[`Popover renders 1`] = `
.c0 {
  right: 0;
  bottom: 0;
  left: 0;
}

<div
  className="c0 "
  open={false}
/>
`;

exports[`Popover renders open 1`] = `
.c0 {
  right: 0;
  bottom: 0;
  left: 0;
  position: fixed;
  z-index: 2;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 background-color:function (props) {
  return: get(props.theme,keys,fallback);
}

@media screen and (min-width:40em) {
  .c0 {
    position: relative;
    top: auto;
    display: block;
  }
}

<div
  className="c0 "
  open={true}
/>
`;
